pyinstrument/profiler.py

```diff
@@ -30,13 +34,21 @@ def __init__(
         start_call_stack: list[str],
         target_description: str,
         interval: float,
+        child_threads: bool
     ) -> None:
         self.start_time = start_time
+        self.thread_start_times = {}
         self.start_process_time = start_process_time
         self.start_call_stack = start_call_stack
         self.frame_records = []
         self.target_description = target_description
-        self.interval = interval
+        self.interval = interval,
+        self.child_threads = child_threads
+
+    def record_thread_start(self, thread_id: str, time: float) -> None:
+        if not self.thread_start_times:
+            self.first_start_time = time
+        self.thread_start_times[thread_id] = time - self.first_start_time
```

Comment:
Why not store the raw start time in each, rather than the offset from the `first_start_time`? Generally I find it better to store truth over computed values.

Reply:
I initially tried raw time but it had some complexity when trying to view the traces on the same view, i'd be happy to look at that again.